docs/spec/v1beta1/gitrepositories.md

```diff
@@ -405,6 +405,35 @@ kubectl create secret generic pgp-public-keys \
     --from-file=author2.asc
 ```
 
+## Self signed certificates
+
+Cloning over HTTPS from a git repository with a self signed certificate:
+
+```yaml
+apiVersion: source.toolkit.fluxcd.io/v1beta1
+kind: GitRepository
+metadata:
+  name: podinfo
+  namespace: default
+spec:
+  interval: 1m
+  url: https://customdomain.com/stefanprodan/podinfo
+  secretRef:
+    name: https-credentials
+  gitImplementation: libgit2
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: https-credentials
+  namespace: default
+type: Opaque
+data:
+  caFile: <CA>
```

Comment:
I would add here username/password and in the note specify that is possible to clone public repos by creating a secret only with `caCert`.